### PR TITLE
Add AIProxy support

### PR DESCRIPTION
**Pull request by lzell**
- Adds instructions for modifying integration code to route requests through AIProxy
- Adds the AIProxy public keys used for TLS handshake (cert pinning)
- Adds the headers and generating code for AIProxy client IDs and DeviceCheck
- Adds service selection screen to the example app:

### CODE REVIEW

File: Sources/Anthropic/Private/Network/Endpoint.swift [hunk 1]
```diff
@@ -27,12 +27,12 @@ protocol Endpoint {
 
 extension Endpoint {
 
-   private func urlComponents(
+   func urlComponents(
```
Comment (lzell) [inline Sources/Anthropic/Private/Network/Endpoint.swift:30]: @jamesrochabrun This is one of the parts of your core lib that I touched. I am re-using this `urlComponents` helper from `Endpoint+AIProxy.swift`

File: Sources/Anthropic/Private/Network/Endpoint.swift [hunk 2]
```diff
@@ -27,12 +27,12 @@ protocol Endpoint {
...
       queryItems: [URLQueryItem])
       -> URLComponents
    {
       var components = URLComponents(string: base)!
-      components.path = path
+      components.path = components.path.appending(path)
```
Comment (lzell) [inline Sources/Anthropic/Private/Network/Endpoint.swift:35]: This approach also works for base URLs that don't include a path in them (e.g. `https://api.anthropic.com`), so existing functionality of the lib is unchanged

Edit to add: FYI AIProxy base urls contain a path already, which is why this change was needed. We call it a `serviceURL`, and it looks like this, for example: `https://api.aiproxy.pro/07b2a53/690171494`

File: Sources/Anthropic/Public/Parameters/Message/MessageParameter.swift [hunk 3]
```diff
@@ -194,7 +194,7 @@ public struct MessageParameter: Encodable {
       public let name: String
       /// A description of what the function does, used by the model to choose when and how to call the function.
       public let description: String?
-      /// The parameters the functions accepts, described as a JSON Schema object. See the [guide](https://platform.openai.com/docs/guides/gpt/function-calling) for examples, and the [JSON Schema reference](https://json-schema.org/understanding-json-schema) for documentation about the format.
+      /// The parameters the functions accepts, described as a JSON Schema object. See the [guide](https://docs.anthropic.com/en/docs/build-with-claude/tool-use) for examples, and the [JSON Schema reference](https://json-schema.org/understanding-json-schema) for documentation about the format.
```
Comment (lzell) [inline Sources/Anthropic/Public/Parameters/Message/MessageParameter.swift:197]: Touched this up to point to the right tool guide

File: Sources/Anthropic/Service/AnthropicServiceFactory.swift [hunk 4]
```diff
@@ -25,7 +25,7 @@ public final class AnthropicServiceFactory {
       basePath: String = "https://api.anthropic.com",
       configuration: URLSessionConfiguration = .default,
       debugEnabled: Bool = false)
-      -> some AnthropicService
+      -> AnthropicService
```
Comment (lzell) [inline Sources/Anthropic/Service/AnthropicServiceFactory.swift:28]: This is another change to the core that I touched. You don't need the `some` keyword here because the protocol that our concrete types are conforming to is not a PAT. That is, we don't need to use 'reverse generics' and can instead just rely on vanilla polymorphism through protocols. We could also make this change in SwiftOpenAI

File: Sources/Anthropic/Public/Parameters/Message/MessageParameter.swift (hunk 3 above)
Comment (jamesrochabrun) [reply Sources/Anthropic/Public/Parameters/Message/MessageParameter.swift:197]: great catch, prevoius copy 🍝

**Review by jamesrochabrun — APPROVED** [review]
Awesome!
